### PR TITLE
Refactor to use google/wire instead of fx

### PR DESCRIPTION
**Pull request by naveensrinivasan**
## Release Notes

- **New Features**
	- Enhanced server configuration options with new storage type and address fields.
	- Introduced dependency injection for server command initialization using Google Wire.
  
- **Bug Fixes**
	- Improved error handling during Redis client initialization.
  
- **Documentation**
	- Added detailed command descriptions for better user guidance.
  
- **Tests**
	- Added comprehensive unit tests for the ingest and command functionality.
  
- **Chores**
	- Updated dependency management, removing unused libraries and adding Google Wire.
  
- **Refactor**
	- Streamlined command initialization by removing unnecessary parameters and dependencies.
	- Simplified data loading functions by eliminating redundant storage parameters.
	- Enhanced build process with new targets and improved error handling.
	- Updated workflow configurations for broader trigger conditions and standardized formatting.

### CODE REVIEW

File: cmd/query/custom/custom.go [hunk 1]
```diff
@@ -1,157 +1,133 @@
...
 		return fmt.Errorf("query failed: %v", err)
 	}
 
-	// Initialize the table
-	table := tablewriter.NewWriter(os.Stdout)
-	table.SetAutoWrapText(false)
-	table.SetRowLine(true)
+	if len(res.Msg.Nodes) == 0 {
+		return fmt.Errorf("no nodes found for script: %s", script)
+	}
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/query/custom/custom.go:65]: I do not want to set timeout as some of these commands will take longer than 30 seconds.

File: cmd/root/root.go [hunk 2]
```diff
@@ -1,41 +1,50 @@
...
+	PprofAddr    string
+	PprofEnabled bool
 }
 
-func (o *options) AddFlags(cmd *cobra.Command) {
-	cmd.PersistentFlags().BoolVar(&o.pprofEnabled, "pprof", false, "enable pprof server")
-	cmd.PersistentFlags().StringVar(&o.pprofAddr, "pprof-addr", "localhost:6060", "address for pprof server")
+func (o *Options) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().BoolVar(&o.PprofEnabled, "pprof", false, "Enable pprof server")
+	cmd.PersistentFlags().StringVar(&o.PprofAddr, "pprof-addr", "localhost:6060", "Address for pprof server")
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/root/root.go:22]: Ignore this as pprof is not enabled by default.

File: cmd/ingest/ingest.go [hunk 3]
```diff
@@ -13,16 +12,16 @@
...
 		SilenceUsage:      true,
 		DisableAutoGenTag: true,
 	}
 
-	cmd.AddCommand(osv.New(storage))
-	cmd.AddCommand(sbom.New(storage))
-	cmd.AddCommand(scorecard.New(storage))
+	cmd.AddCommand(osv.New())
+	cmd.AddCommand(sbom.New())
+	cmd.AddCommand(scorecard.New())
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/ingest/ingest.go:25]: Yes please

File: cmd/leaderboard/leaderboard.go [hunk 4]
```diff
@@ -3,15 +3,14 @@
...
-	"github.com/bitbomdev/minefield/pkg/graph"
 	"github.com/spf13/cobra"
 )
 
 type options struct{}
 
 func (o *options) AddFlags(_ *cobra.Command) {}
 
-func New(storage graph.Storage) *cobra.Command {
+func New() *cobra.Command {
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/leaderboard/leaderboard.go:13]: Only the server has wire configuration, and the rest of them don't have any wire configuration as of now.

File: cmd/ingest/osv/osv.go [hunk 5]
```diff
@@ -37,7 +35,7 @@
 	}
 	vulnsPath := args[0]
 	// Ingest vulnerabilities
-	result, err := ingest.LoadDataFromPath(o.storage, vulnsPath)
+	result, err := ingest.LoadDataFromPath(vulnsPath)
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/ingest/osv/osv.go:38]: Yes, please, and a a reference to this.

File: cmd/server/wire_gen.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	storage, err := ProvideStorage(o)
+	if err != nil {
+		return nil, err
+	}
+	command, err := NewServerCommand(storage, o)
+	if err != nil {
+		return nil, err
+	}
+	return command, nil
+}
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/server/wire_gen.go:26]: Add it to the backlog please

[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/server/wire_gen.go:26]: Yes, please

File: cmd/root/root.go [hunk 7]
```diff
@@ -1,41 +1,59 @@
...
-	cmd.AddCommand(leaderboard.New(storage))
-	cmd.AddCommand(server.New(storage))
+	rootCmd.AddCommand(query.New())
+	rootCmd.AddCommand(ingest.New())
+	rootCmd.AddCommand(cache.New())
+	rootCmd.AddCommand(leaderboard.New())
+	rootCmd.AddCommand(server.New())
 
-	return cmd
+	return rootCmd
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply cmd/root/root.go:58]: Add a GitHub Issue.

**Review by neilnaveen — APPROVED** [review]

**Pull request merged**